pylint/lint.py

```diff
@@ -946,12 +946,20 @@ def register_checker(self, checker):
 
         :raises ValueError: If the priority of the checker is invalid.
         """
-        if checker.name in self._checkers:
-            # TODO: Raise if classes are the same
-            for duplicate in self._checkers[checker.name]:
-                msg = 'A checker called {} has already been registered ({}).'
-                msg = msg.format(checker.name, duplicate.__class__)
-                warnings.warn(msg)
+        existing_checker_types = set(
+            type(existing_checker)
+            for name_checkers in self._checkers.values()
+            for existing_checker in name_checkers
+        )
+        checker_type = type(checker)
+        if checker_type in existing_checker_types:
+            msg_fmt = (
+                'Not registering checker {}. A checker of type {} has '
+                'already been registered.'
+            )
+            msg = msg_fmt.format(checker.name, checker_type.__name__)
+            warnings.warn(msg)
```

Comment:
Keeping a warning, but not raising. Can remove warning if not needed.

Reply:
I notice you removed the warning (and checker code) before, but keeping the warning here now as there is a check again and I got a thumbs up :)